lib/date_range.rb

```diff
@@ -0,0 +1,38 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :check_in, :check_out
+    
+    def initialize(check_in, check_out)
+      @check_in = check_in
+  
+      if check_out < check_in
```

Comment:
You're checking this twice, here and with the `total_stay` method.  A little odd.  Also this if statement doesn't check to see if the dates are equal, but your total_stay method did.